### PR TITLE
refactor(sbom): add sbom prefix + filepaths for decode log messages

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
Add `file_path` + `sbom` prefix for log messages in `decode.go`.

before:
```bash
2024-07-02T16:06:42+06:00       WARN    Ignore the OS package as no OS is detected.
```

after: 
```bash
2024-07-02T14:44:45+06:00       WARN    [sbom] Ignore the OS package as no OS is detected.      file_path="var/lib/db/sbom/apk-tools-2.14.4-r0.spdx.json"
```

## Related discussions
- Close #7073

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [x] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/fanal/analyzer/sbom/sbom.go [hunk 1]
```diff
@@ -29,14 +30,15 @@ var requiredSuffixes = []string{
...
-func (a sbomAnalyzer) Analyze(_ context.Context, input analyzer.AnalysisInput) (*analyzer.AnalysisResult, error) {
+func (a sbomAnalyzer) Analyze(ctx context.Context, input analyzer.AnalysisInput) (*analyzer.AnalysisResult, error) {
 	// Format auto-detection
 	format, err := sbom.DetectFormat(input.Content)
 	if err != nil {
 		return nil, xerrors.Errorf("failed to detect SBOM format: %w", err)
 	}
 
-	bom, err := sbom.Decode(input.Content, format)
+	ctx = log.WithContextAttrs(ctx, log.String("file", input.FilePath))
```
Comment (knqyf263) [inline pkg/fanal/analyzer/sbom/sbom.go:40]: What if adding `log.FilePath(filePath string)` in `pkg/log` and using it so we can use the consistent key name?

```
func FilePath(filePath string) slog.Attr {
    return String("file_path", filePath)
}
```

File: pkg/sbom/spdx/unmarshal_test.go [hunk 2]
```diff
@@ -357,7 +359,8 @@ func TestUnmarshaler_Unmarshal(t *testing.T) {
 			}
 
 			var got types.SBOM
-			err = sbomio.NewDecoder(v.BOM).Decode(&got)
+			ctx := log.WithContextAttrs(context.Background(), log.String("filepath", tt.inputFile))
+			err = sbomio.NewDecoder(v.BOM).Decode(ctx, &got)
```
[the remark this answers is not in the document]
Comment (DmitriyLewen) [reply pkg/sbom/spdx/unmarshal_test.go]: removed in https://github.com/aquasecurity/trivy/pull/7074/commits/0e35db078a3b1adf09930b513d05b3fdefee3fc1

File: pkg/sbom/cyclonedx/unmarshal_test.go [hunk 3]
```diff
@@ -757,7 +759,8 @@ func TestUnmarshaler_Unmarshal(t *testing.T) {
 			require.NoError(t, err)
 
 			var got types.SBOM
-			err = sbomio.NewDecoder(cdx.BOM).Decode(&got)
+			ctx := log.WithContextAttrs(context.Background(), log.String("filepath", tt.inputFile))
+			err = sbomio.NewDecoder(cdx.BOM).Decode(ctx, &got)
```
[the remark this answers is not in the document]
Comment (DmitriyLewen) [reply pkg/sbom/cyclonedx/unmarshal_test.go]: removed https://github.com/aquasecurity/trivy/pull/7074/commits/0e35db078a3b1adf09930b513d05b3fdefee3fc1

File: pkg/fanal/analyzer/sbom/sbom.go (hunk 1 above)
Comment (DmitriyLewen) [reply pkg/fanal/analyzer/sbom/sbom.go]: added (https://github.com/aquasecurity/trivy/pull/7074/commits/5f3f694ca6b2f39e09916a66ba6a22b4293fe453 + https://github.com/aquasecurity/trivy/pull/7074/commits/ae1e4d818c70b65517f5040ffa0f4ad29ba7dd8e)

Comment (knqyf263) [reply pkg/fanal/analyzer/sbom/sbom.go]: >added (https://github.com/aquasecurity/trivy/commit/5f3f694ca6b2f39e09916a66ba6a22b4293fe453 + https://github.com/aquasecurity/trivy/commit/ae1e4d818c70b65517f5040ffa0f4ad29ba7dd8e)

I think they should be another PR. And merge the change into this PR.

Comment (DmitriyLewen) [reply pkg/fanal/analyzer/sbom/sbom.go]: You are right. The number of filepaths used was more than i expected.
I created #7080

Comment (knqyf263) [reply pkg/fanal/analyzer/sbom/sbom.go]: Thanks.  Merged #7080.

Comment (DmitriyLewen) [reply pkg/fanal/analyzer/sbom/sbom.go]: updated this PR

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**